src/Listeners/Plugin.php

```diff
@@ -44,7 +56,212 @@ public function deactivated( $plugin, $network_wide ) {
 		$data = array(
 			'plugin'       => $plugin,
 			'network_wide' => $network_wide,
+			'plugins'      => $this->collect_plugin( $plugin, false ),
 		);
 		$this->push( 'plugin_deactivated', $data );
 	}
+
+	/**
+	 * Plugin deleted
+	 *
+	 * @param string  $plugin Name of the plugin
+	 * @param boolean $deleted Whether the plugin deletion was successful
+	 * @return void
+	 */
+	public function deleted( $plugin, $deleted ) {
+		// abort if not successfully deleted
+		if ( !$deleted ) {
+			return;
+		}
+		
+		$slug = self::glean_plugin_slugname( $plugin );
+		$data = array(
+			'plugin' => $slug,
+		);
+		$this->push( 'plugin_deleted', $data );
+	}
+
+	/**
+	 * Plugin install or update completed
+	 *
+	 * @param string  $upgrader_object Upgrader Object from upgrade hook
+	 * @param boolean $options Options from upgrade hook including type, action & plugins.
+	 * @return void
+	 */
+	public function updated( $upgrader_object, $options ) {
+		// bail if not plugin install or update
+		if ( 'plugin' !== $options['type'] ) {
+			return;
+		}
+		// set event type
+		if ( 'update' === $options['action'] ) {
+			$event_key = 'plugin_updated';
+
+			$plugins = [];
+			if ( isset( $options['plugins'] ) && is_array( $options['plugins'] ) ) {
+				foreach ( $options['plugins'] as $index => $pluginslug ) {
+					$plugin = $this->collect_plugin( 
+						$pluginslug, 
+						is_plugin_active( $pluginslug ),
+						false
+					);
+					array_push( $plugins, $plugin );
+				}
+			}
+			$data = array(
+				'plugins' => $plugins,
+			);
+		
+		} elseif ('install' === $options['action'] ) {
+			$event_key = 'plugin_installed';
+			// get all plugins - slug not returned for install actions
+			$plugins = $this->collect_plugins();
+			$data = array(
+				'plugins' => $plugins,
+			);
+
+		} else {
+			return;
+		}
+		
+		$this->push( $event_key, $data );
+	}
+
+	/**
+	 * Grab relevant data from plugin data - and only what we want
+	 * @param Array $data The plugin meta data from the header
+	 * @return Array Hiive relevant plugin details
+	 */
+	public static function glean_plugin_data( $data ){
+		$plugin = [];
+		$plugin['version'] = $data['Version'];
+		$plugin['title'] = $data['Name'] ? $data['Name'] : '';
+		$plugin['url'] = $data['PluginURI'] ? $data['PluginURI'] : '';
+		return $plugin;
+	}
+
+	/**
+	 * Get plugin dir name or main file if no dir. 
+	 * @param string  $slug Relative path to plugin file
+	 * @return string slug for plugin
+	 */
+	public static function glean_plugin_slugname( $slug ){
+		$newslug = '';
+		// if has `/` split on `/` and get second to last index.
+		if ( str_contains( $slug, '/' ) ) {
+			$parts = explode( '/', $slug );
+			end( $parts );
+			$newslug = prev( $parts );
+		} else {
+			// if not, split on . and get second to last index.
+			$parts = explode( '.', $slug );
+			end( $parts );
+			$newslug = prev( $parts );
+		}
+		return $newslug;
+	}
```

Comment:
Does this mean we're sending (and storing on the Hiive) just the plugin directory instead of the standard full slug?

Reply:
Yes, it's set up to grab the folder name or the file name if there is no folder. This combines things like `hello-dolly/hello-dolly.php` into a simple `hello-dolly`. To remove, we'd end up with `bluehost-wordpress-plugin/bluehost-wordpress-plugin.php` which is rather long in place of ` bluehost-wordpress-pluign`. Looking at it again now though, it may be safer to have the full slug as WP sees it. What do you think?

Reply:
Using the longer wp slug for now. This is updated.